sydent/threepid/bind.py

```diff
@@ -40,13 +40,30 @@
 
 logger = logging.getLogger(__name__)
 
+def parseMxid(mxid):
+    if len(mxid) > 255:
+        raise Exception("This mxid is too long")
+
+    if len(mxid) == 0 or mxid[0:1] != "@":
+        raise Exception("mxid does not start with '@'")
+
+    parts = mxid[1:].split(':', 1)
+    if len(parts) != 2:
```

Comment:
IIRC spec says that a MXID can also have the form `@localpart:host:port` (since a server name can be `host:port`), so shouldn't this be `if len(parts) < 2:`?
Not that I think it matters a lot in this specific case.

Reply:
mxids can also contain ipv6 literals, so you can have Many colons...

Reply:
The split is limited to one split, so there can't be more than two parts here.

Reply:
Oh, right, I missed that.